src/components/REPLInput.js

```diff
@@ -33,10 +40,22 @@ export default class REPLInput extends React.Component {
           {'> '}
         </span>
         <input style={styles.input}
+               type="text"
                value={this.props.code}
                onChange={event => this.props.changeREPLCode(event.target.value)}
                onKeyPress={event =>
-                 event.key === "Enter" && this.props.onRun(this.props.code)} />
+                 event.key === "Enter" && this.props.onRun(this.props.code)}
+               onKeyDown={event => {
+                 if (event.key === "ArrowUp") {
+                   this.props.prevREPLCode();
+                   this.props.displayNewREPLHistoryCode();
+                 } else {
+                   if (event.key === "ArrowDown") {
+                     this.props.nextREPLCode();
+                     this.props.displayNewREPLHistoryCode();
```

Comment:
because in order to access the previous code codeIndex must first be updated. I tried to use merge to update both the code index and the display code in just the single case statement, but I kept receiving "undefined" as the code being accessed, which made me think it was an index error. After looking through the state (this was a week ago so I don't fully remember all the details) I came to the conclusion that they had to be dispatched in separate actions so that codeIndex would be explicitly updated before trying to access the history code. Do you think the error might have been caused by something else other than the index error?